relay/chunkstore/chunk_reader.go

```diff
@@ -71,30 +91,109 @@ func (r *chunkReader) GetBinaryChunkProofs(ctx context.Context, blobKey corev2.B
 func (r *chunkReader) GetBinaryChunkCoefficients(
 	ctx context.Context,
 	blobKey corev2.BlobKey,
-	fragmentInfo *encoding.FragmentInfo) (uint32, [][]byte, error) {
+) (uint32, [][]byte, error) {
 
-	bytes, err := r.client.FragmentedDownloadObject(
-		ctx,
-		r.bucket,
-		s3.ScopedChunkKey(blobKey),
-		int(fragmentInfo.TotalChunkSizeBytes),
-		int(fragmentInfo.FragmentSizeBytes))
+	bytes, found, err := r.client.DownloadObject(ctx, r.bucket, s3.ScopedChunkKey(blobKey))
+	if !found {
+		return 0, nil, fmt.Errorf("coefficients not found for blob %s", blobKey.Hex())
+	}
 
 	if err != nil {
-		r.logger.Error("failed to download coefficients from S3",
-			"blob", blobKey.Hex(),
-			"totalSize", fragmentInfo.TotalChunkSizeBytes,
-			"fragmentSize", fragmentInfo.FragmentSizeBytes,
-			"error", err)
-		return 0, nil, fmt.Errorf("failed to download coefficients from S3 for blob %s (total size: %d, fragment size: %d): %w",
-			blobKey.Hex(), fragmentInfo.TotalChunkSizeBytes, fragmentInfo.FragmentSizeBytes, err)
+		return 0, nil, fmt.Errorf("failed to download coefficients from S3 for blob %s: %w", blobKey.Hex(), err)
 	}
 
 	elementCount, frames, err := rs.SplitSerializedFrameCoeffs(bytes)
 	if err != nil {
-		r.logger.Error("failed to split coefficient frames", "blob", blobKey.Hex(), "error", err)
 		return 0, nil, fmt.Errorf("failed to split coefficient frames for blob %s: %w", blobKey.Hex(), err)
 	}
 
 	return elementCount, frames, nil
 }
+
+// TODO handle not found error properly!!!
+
+func (r *chunkReader) GetBinaryChunkProofsRange(
+	ctx context.Context,
+	blobKey corev2.BlobKey,
+	startIndex uint32,
+	endIndex uint32,
+) ([][]byte, bool, error) {
+
+	if startIndex >= endIndex {
+		return nil, false, fmt.Errorf("invalid startIndex (%d) or endIndex (%d)", startIndex, endIndex)
+	}
+
+	firstByteIndex := startIndex * encoding.SerializedProofLength
+	count := endIndex - startIndex
+	size := count * encoding.SerializedProofLength
+
+	s3Key := s3.ScopedProofKey(blobKey)
+
+	data, found, err := r.client.DownloadPartialObject(
+		ctx,
+		r.bucket,
+		s3Key,
+		int64(firstByteIndex),
+		int64(firstByteIndex+size))
```

Comment:
seems like you're doing a lot of gymnastics to compute this value, instead of just using `endIndex` passed in as a parameter

same comment applies for `GetBinaryChunkCoefficientRange` method below